hbase-client/src/main/java/org/apache/hadoop/hbase/client/Admin.java

```diff
@@ -705,6 +705,17 @@ void majorCompact(TableName tableName, byte[] columnFamily, CompactType compactT
   Map<ServerName, Boolean> compactionSwitch(boolean switchState, List<String> serverNamesList)
       throws IOException;
 
+  /**
+   * Turn the compaction offload on or off. This state is ephemeral. The setting will be lost on
+   * restart.
+   * @param switchState Set to <code>true</code> to enable, <code>false</code> to disable.
+   * @param serverNamesList list of region servers.
+   * @return Previous compaction offload states for region servers
+   * @throws IOException if a remote or network exception occurs
+   */
+  Map<ServerName, Boolean> compactionOffloadSwitch(boolean switchState, List<String> serverNamesList)
```

Comment:
Why adding a separate API? We'll either have embedded or offload compaction configured to run, so the existing `compactionSwitch` command should transparently switch on/off whatever compaction execution strategy is in place.

Reply:
This api can change compaction execution strategy for specific regionservers conveniently. 
It's useful, isn't it ?